File: Data Structures & Algorithms/binary-search/submission-13.cs
public class Solution {
    public int Search(int[] nums, int target) {
        
        int l = 0; 
        int r = nums.Length - 1; 

        while (l <= r){
            int mid = l + (r - l) / 2 ;

            if(nums[mid] == target) return mid;

            if(nums[mid] > target){
                r --;
            }else{
                l ++;
            }
        }

        return -1;
    }
}

File: Data Structures & Algorithms/top-k-elements-in-list/submission-13.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        map = {}

        for i in nums:
            if i in map.keys():
                map[i] += 1
            else:
                map[i] = 1

        heap = []
        for key in map.keys():
            heapq.heappush(heap, (map[key],key))

            if len(heap) > k:
                heapq.heappop(heap)
            
        res = []

        for r in range(k):
            print(r)
            res.append(heapq.heappop(heap)[1])

        return res